### PR TITLE
Refactor to ease definition display-type specific colors

### PR DESCRIPTION
**Pull request by eigengrau**
This refactors the theme so that the primary color table can specify alternative
colors to be used in the terminal. This is meant to be of use when Emacs doesn’t
do a sufficiently good job at mapping the theme colors to the 256-XTerm palette.

### CODE REVIEW

**Comment by eigengrau** [comment]
So, I’m not sure whether we could do a much better job than Emacs at mapping the theme to the XTerm palette, but I figured we might give it a shot. This PR doesn’t change any of the colors though, so the only tty-specific color is `bg1` for now.

**Comment by zenorocha** [comment]
What do you think @film42?

**Comment by film42** [comment]
I think I'm a bit unqualified to review this one, but making things more customizable seems like a good idea. There are one or two other small changes that we should probably get in before merging this one since a rebase will be required.

**Comment by eigengrau** [comment]
Rebased.

File: dracula-theme.el [hunk 1]
```diff
@@ -17,411 +17,420 @@
...
+               ;; rpm-spec
+               (rpm-spec-dir-face :foreground ,rainbow-6)
+               (rpm-spec-doc-face :foreground ,rainbow-4)
+               (rpm-spec-ghost-face :foreground ,rainbow-3)
+               (rpm-spec-macro-face :foreground ,rainbow-7)
+               (rpm-spec-obsolete-tag-face :inherit font-lock-warning-face)
+               (rpm-spec-package-face :foreground ,rainbow-3)
+               (rpm-spec-section-face :foreground ,rainbow-7)
+               (rpm-spec-tag-face :foreground ,rainbow-2)
+               (rpm-spec-var-face :foreground "#a0522d")
```
Comment (film42) [inline dracula-theme.el]: Would you mind giving this custom color a label and moving it up?

File: dracula-theme.el [hunk 2]
```diff
@@ -17,411 +17,420 @@
...
+               (outline-5 :foreground ,rainbow-5)
+               (outline-6 :foreground ,rainbow-8)
+               ;; powerline
+               (powerline-evil-base-face :foreground ,bg2)
+               (powerline-evil-emacs-face :inherit powerline-evil-base-face :background ,rainbow-7)
+               (powerline-evil-insert-face :inherit powerline-evil-base-face :background ,rainbow-2)
+               (powerline-evil-motion-face :inherit powerline-evil-base-face :background ,rainbow-3)
+               (powerline-evil-normal-face :inherit powerline-evil-base-face :background ,rainbow-6)
+               (powerline-evil-operator-face :inherit powerline-evil-base-face :background ,rainbow-4)
+               (powerline-evil-replace-face :inherit powerline-evil-base-face :background "#ff5555")
```
Comment (film42) [inline dracula-theme.el]: Would you mind giving this custom color a label and moving it up?

File: dracula-theme.el (hunk 1 above)
Comment (eigengrau) [reply dracula-theme.el]: @film42 These face specs are unchanged from the current git HEAD, just moved around. We can change the PR into a generic refactoring PR if you prefer, though. You should be able to push to my `refactor` branch if you find anything you want to address in addition to tty-specific colors.

I’ll commit the changes you’ve suggested so far.

**Comment by eigengrau** [comment]
I noticed one of the hex-coded colors already had a label. For the other I’ve added `rainbow-10`, though I’m not sure it’s prudent, since it may encourage folks to use that color in future changes and it’s not such a great fit with the rest of the scheme. Not sure why it was originally chosen.

**Comment by film42** [comment]
Thanks @eigengrau ! 

Agree it's kind of a weird color, but since it's already in place might be best to leave it for now and we can re-evaluate later. At least it's now easier to change.

**Pull request merged**